subsys/shell/shell_ops.h

```diff
@@ -25,14 +25,12 @@ static inline void z_shell_raw_fprintf(const struct shell_fprintf *const ctx,
 	va_end(args);
 }
 
-/* Macro to send VT100 commands. */
+/* Macro to send VT100 command. */
 #define Z_SHELL_VT100_CMD(_shell_, _cmd_)				\
 	do {								\
 		if (!IS_ENABLED(CONFIG_SHELL_VT100_COMMANDS))		\
 			break;						\
-									\
-		static const char cmd[] = _cmd_;			\
-		z_shell_raw_fprintf(_shell_->fprintf_ctx, "%s", cmd);	\
+		z_shell_raw_fprintf(_shell_->fprintf_ctx, "%s", _cmd_);	\
```

Comment:
have you consider using _cmd_ directly
`z_shell_raw_fprintf(_shell_->fprintf_ctx,  __VA_ARGS__);`

Then for color you could have something like
`#define VT100_BG_COLOR \e[403%cm`
and call
`Z_SHELL_VT100_CMD(sh, VT100_BG_COLOR, c + 0x30)`